RPED Patches

SyncIt21 opened the pull request:
## About The Pull Request
Fixes #71801 & also merges changes from  #71775

## Why It's Good For The Game
Bugs. We all hate them & yet we are very good at making them PAIN!!!

## Changelog
:cl:
fix: RPED part display information
fix: RPED incorrectly exchanging parts with the wrong tier for datum stock parts
fix: RPED unable to install datum stock parts in incomplete machine frames
fix: RPED unable to exchange parts for newly constructed machines
/:cl:

SyncIt21 committed:
RPED Patches

1. Display correct number of datum stock parts
2. First sorts parts before swaping them
3. Extra type check for if the path is datum or obj

SyncIt21 commented:
@Mothblocks 👍

Mothblocks commented on code/modules/research/stock_parts.dm:
```diff
@@ -216,6 +216,15 @@ If you create T5+ please take a pass at mech_fabricator.dm. The parts being good
+/obj/item/storage/part_replacer/proc/get_sorted_parts()
+	var/list/part_list = list()
+	//Assemble a list of current parts, then sort them by their rating!
+	for(var/obj/item/co in contents)
```
Full name--`component`

SyncIt21 replied:
Done

SyncIt21 committed:
Full name

SyncIt21 committed:
Merge branch 'master' into RPED-Fixes

Mothblocks commented on code/game/machinery/_machinery.dm:
```diff
@@ -930,6 +930,13 @@
 		return FALSE
 
+	/**
+	 * sorting is very important especially because we are breaking the loop on line 1006
```
Please point this to the specific piece of the proc, the lines are going to change.

SyncIt21 replied:
Done

SyncIt21 committed:
Correct comments

SyncIt21 committed:
Grammer

SyncIt21 committed:
More bug fixes

1. adds circuit board to the list of components so part_replacer can locate it
2. refractors for loop for deconstruct

SyncIt21 commented:
@Mothblocks RPED was not exchanging parts for newly constructed machines because circuit board was not added to the list of components. Thus exchange_parts() was unable to locate it and did an early return fixed that now.
Also another bug was that if an device required say 2 manipulators and if the RPED only had one the machine the RPED would not remove that installed part from the sorted list after installing causing that one manipulator to replace both in the machine. Thats also fixed now

SyncIt21 committed:
datum

SyncIt21 committed:
datum & obj
